x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -223,6 +223,19 @@ export class ExecutionHandler<
           continue;
         }
 
+        if (await this.actionsClient!.hasReachedTheQueuedActionsLimit()) {
```

Comment:
If a rule triggers a lot of actions, we will be performing a query against the task manager index in each iteration of this for loop. Are there any concerns about performance in this scenario?

Reply:
I am not sure, the query is not very big. Do you have any ideas for improving the performance? We can run the query once, and then check number of actions that the rule wants to trigger against the number queued?

Reply:
Resolved in this commit [cc92321](https://github.com/elastic/kibana/pull/164632/commits/cc923210bbc901229372ad2c66119b4b5f1354a8)